samples/net/mqtt_publisher/src/main.c

```diff
@@ -248,6 +251,13 @@ static void broker_init(void)
 	broker4->sin_family = AF_INET;
 	broker4->sin_port = htons(SERVER_PORT);
 	inet_pton(AF_INET, SERVER_ADDR, &broker4->sin_addr);
+#if defined(CONFIG_SOCKS) && !defined(CONFIG_MQTT_LIB_TLS)
```

Comment:
can you use IS_ENABLED() etc...

Than will also make below proxy variable declared at the beginning of a code block, which would be proper unlike what it is right now.